src/common/src/block_builder.cc

```diff
@@ -0,0 +1,147 @@
+#include "block_builder.h"
+#include <bits/stdc++.h>
+
+BlockBuilder::BlockBuilder(void)
+{
+}
+
+BlockBuilder::BlockBuilder(const std::vector<std::vector<std::string> > &records)
+{
+    for (size_t i = 1; i < records.size(); ++i)
```

Comment:
You can also iterate over the records with `for (const std::vector<std::string>> &record : records)`.  It's fine if you don't make this change, but in that case, please change auto to the actual type.

Reply:
Iteration start at the second row because the first row is just field names so we want to skip it. 
line 12: `const auto` changed to `std::vector<std::string>`